io/read.go

```diff
@@ -53,3 +55,14 @@ func consistentReadSync(filename string, attempts int, sync func(int)) ([]byte,
 	}
 	return nil, fmt.Errorf("could not get consistent content of %s after %d attempts", filename, attempts)
 }
+
+// ReadAtMost reads up to `limit` bytes from `r`, and reports an error
+// when `limit` bytes are read.
+func ReadAtMost(r io.Reader, limit int64) ([]byte, error) {
+	limitedReader := &io.LimitedReader{R: r, N: limit}
+	data, err := ioutil.ReadAll(limitedReader)
+	if limitedReader.N <= 0 {
+		return data, errors.New("the read limit is reached")
```

Comment:
I think the error should be defined as an exported global - see https://github.com/kubernetes/kubernetes/pull/76518#discussion_r277083231

Reply:
Fixed! Thanks!